bpf/lib/nodeport.h

```diff
@@ -3063,17 +3063,23 @@ nodeport_rev_dnat_fwd_ipv4(struct __ctx_buff *ctx, bool *snat_done,
 		return CTX_ACT_OK;
 
 #if defined(IS_BPF_HOST) && !defined(ENABLE_SKIP_FIB)
-	/* Perform FIB lookup with post-RevDNAT src IP, and redirect
-	 * packet to the correct egress interface:
-	 */
-	fib_params.l.family = AF_INET;
-	fib_params.l.ifindex = ctx_get_ifindex(ctx);
-	fib_params.l.ipv4_src = nat_info->address;
-	fib_params.l.ipv4_dst = tuple.daddr;
+# if defined(ENABLE_WIREGUARD)
+	if (ctx_get_ifindex(ctx) != WG_IFINDEX) {
+# endif /* ENABLE_WIREGUARD */
+		/* Perform FIB lookup with post-RevDNAT src IP, and redirect
+		 * packet to the correct egress interface:
+		 */
+		fib_params.l.family = AF_INET;
+		fib_params.l.ifindex = ctx_get_ifindex(ctx);
+		fib_params.l.ipv4_src = nat_info->address;
+		fib_params.l.ipv4_dst = tuple.daddr;
```

Comment:
Have you considered adding a new `bpf_wireguard.c` file, so that the program can identify itself as `IS_BPF_WIREGUARD` instead?

Reply:
Yeah definitely, I was just not sure which one is better. Currently busy with debugging broken NS connectivity :fearful:

Reply:
It turns out the skbs are entering tail_handle_snat_fwd_ipv4 tailcall from tail_handle_nat_fwd_ipv4 after failing rev DNAT. For now I added an early return after rev dnat:

```c
static __always_inline int
__handle_nat_fwd_ipv6(struct __ctx_buff *ctx, struct trace_ctx *trace,
		      __s8 *ext_err)
{
	bool snat_done = false;
	int ret;

	ret = nodeport_rev_dnat_fwd_ipv6(ctx, &snat_done, trace, ext_err);
	if (ret != CTX_ACT_OK)
		return ret;

+ #if defined(ENABLE_WIREGUARD)
+ 	if (ctx_get_ifindex(ctx) == WG_IFINDEX) {
+		return CTX_ACT_OK;
+	}
+ #endif /* ENABLE_WIREGUARD */
```

If CI is happy with this, I'll then move the whole thing into a `bpf_wg.c` with specific macros.